Source/JavaScriptCore/wasm/WasmFunctionParser.h

```diff
@@ -3011,7 +3011,7 @@ FOR_EACH_WASM_MEMORY_STORE_OP(CREATE_CASE)
         args.grow(exceptionSignature.argumentCount());
         for (unsigned i = 0; i < exceptionSignature.argumentCount(); ++i) {
             TypedExpression arg = m_expressionStack.at(m_expressionStack.size() - i - 1);
-            WASM_VALIDATOR_FAIL_IF(arg.type() != exceptionSignature.argumentType(exceptionSignature.argumentCount() - i - 1), "The exception being thrown expects the argument at index ", i, " to be ", exceptionSignature.argumentType(exceptionSignature.argumentCount() - i - 1), " but argument has type ", arg.type());
+            WASM_VALIDATOR_FAIL_IF(!isSubtype(arg.type(), exceptionSignature.argumentType(exceptionSignature.argumentCount() - i - 1)), "The exception being thrown expects the argument at index ", i, " to be ", exceptionSignature.argumentType(exceptionSignature.argumentCount() - i - 1), " but argument has type ", arg.type());
```

Comment:
This won't change the behavior of non-gc types, right?

Reply:
It shouldn't, in that if neither GC nor typed funcrefs is turned on then `isSubtype` will eventually fallback to just type equality. (https://searchfox.org/wubkat/source/Source/JavaScriptCore/wasm/WasmFormat.h#338)

If we want to be careful, we could adjust `isSubtype` to explicitly check `Options::useWebAssemblyTypedFunctionReferences` and only do type equality if the flag is off.